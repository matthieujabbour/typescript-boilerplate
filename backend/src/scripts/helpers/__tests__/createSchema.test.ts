import createSchema from 'scripts/helpers/createSchema';

describe('helpers/createSchema', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('with body', async () => {
    expect(createSchema({ body: { required: ['test'] } })).toMatchSnapshot();
  });

  test('with query', async () => {
    expect(createSchema({ query: { required: ['test'] } })).toMatchSnapshot();
  });

  test('with params', async () => {
    expect(createSchema({ params: { required: ['test'] } })).toMatchSnapshot();
  });
});
