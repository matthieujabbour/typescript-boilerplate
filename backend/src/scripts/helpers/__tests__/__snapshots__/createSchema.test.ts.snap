// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers/createSchema with body 1`] = `
Object {
  "body": Object {
    "additionalProperties": false,
    "errorMessage": Object {
      "type": "Request body should be a valid object.",
    },
    "required": Array [
      "test",
    ],
    "type": "object",
  },
}
`;

exports[`helpers/createSchema with params 1`] = `
Object {
  "params": Object {
    "additionalProperties": false,
    "errorMessage": Object {
      "type": "Request params should be a valid object.",
    },
    "required": Array [
      "test",
    ],
    "type": "object",
  },
}
`;

exports[`helpers/createSchema with query 1`] = `
Object {
  "query": Object {
    "additionalProperties": false,
    "errorMessage": Object {
      "type": "Request query should be a valid object.",
    },
    "required": Array [
      "test",
    ],
    "type": "object",
  },
}
`;
